pallets/erc20-xcm-bridge/src/erc20_matcher.rs

```diff
@@ -67,6 +67,28 @@ impl<Erc20MultilocationPrefix: Get<MultiLocation>> Erc20Matcher<Erc20Multilocati
 			_ => Err(()),
 		}
 	}
+	pub(crate) fn matches_gas_limit(multiasset: &MultiAsset) -> Option<u64> {
```

Comment:
This function is useless and increase the call stack for nothing, please remove it and move it's body inside weight_of_erc20_transfer

Reply:
I think you could use `if let` to write this concisely instead of having a helper fn

Reply:
@notlesh tried to follow your suggestion, let me know what you think about the new implementation [here](https://github.com/moonbeam-foundation/moonbeam/pull/2422/files#diff-60c41c4cef32c59e32153a0cdac0d3c4765c302f0b51e28d1fe7afc0b949f1bbR69)